SharedCode/Repository/DB/PokemonRepositoryLocal.cs

```diff
@@ -0,0 +1,123 @@
+using System.Linq;
+using System.Threading.Tasks;
+using SharedCode.Database;
+using SharedCode.Model;
+using SharedCode.Model.Api;
+using SharedCode.Model.DB;
+using SharedCode.Repository.Interfaces;
+using SharedCode.Util;
+using Unity;
+
+namespace SharedCode.Repository.DB
+{
+	public class PokemonRepositoryLocal : IPokemonRepositoryLocal
+	{
+		private readonly INetworkHandler NetworkHandler;
+
+        public IPokemonRepository Repository;
+
+        [Dependency]
+		public IDatabaseManager DatabaseManager;
+
+        public PokemonRepositoryLocal(IDatabaseManager databaseManager, IPokemonRepository repository)
+		{
+            this.Repository = repository;
+			this.DatabaseManager = databaseManager;
+            //BuildPokemonLocalList();
+		}
+
+		public async Task BuildPokemonLocalList()
+		{
+			if (DatabaseManager.checkTableExists("Pokemon"))
+				return;
+
+			var data = await Repository.GetPokemonList();
+
+			if (data.Success)
+			{
+                Dictionary<int, PokemonLocal> pokemons = data.Value.Select(pokemon => new PokemonLocal(pokemon.name, pokemon.GetIdFromUrl())).ToDictionary(x => x.Id, x => x);
+				var typesList = await Repository.GetPokemonTypesList();
+				if (typesList.Success)
+				{
+                    foreach (var type in typesList.Value)
+					{
+						var typeInfo = await Repository.GetTypeInfo(type.GetIdFromUrl());
+						if (typeInfo.Success)
+						{
+							pokemons = PopulateTypeForPokemons(pokemons, typeInfo.Value.pokemon, typeInfo.Value.name);
+						}
+					}
+					
+                }
+
+				var generationsList = await Repository.GetPokemonGenerationList();
+				if (generationsList.Success)
+				{
+					foreach (var generation in generationsList.Value)
+					{
+						var generationInfo = await Repository.GetGenerationInfo(generation.GetIdFromUrl());
+						if (generationInfo.Success)
+						{
+                            pokemons = PopulateRegionForPokemons(pokemons, generationInfo.Value.pokemon_species, generationInfo.Value.main_region.name);
+						}
+					}
+				}
+
+                foreach (var finalPokemon in pokemons)
+                {
+                    DatabaseManager.StoreData<PokemonLocal>(finalPokemon.Value, "Pokemon");
+                }
+            }
+		}
+
+        public Dictionary<int, PokemonLocal> PopulateTypeForPokemons(Dictionary<int, PokemonLocal> pokemonsList, List<Pokemon> pokemonTypeList, string TypeName)
+		{
+			var pokemonTypeListIndex = 0;
+			var auxPokemonList = pokemonsList;
+			while(pokemonTypeListIndex < pokemonTypeList.Count())
+			{
+                var id = pokemonTypeList[pokemonTypeListIndex].pokemon.GetIdFromUrl();
+                if (id > 1010)
+                {
+                    pokemonTypeListIndex++;
+                    continue;
+                }
+                if (auxPokemonList[id].Types == "")
+                    auxPokemonList[id].Types += TypeName;
+                else
+                    auxPokemonList[id].Types += $"%{TypeName}";
```

Comment:
Just for clarification, pokemon types are being stored in this format: **"typeA%typeB"**, right?

Reply:
thanks